---
layout: about
title: About IQ-TREE
permalink: /about/
---

<div class="container">
    <div class="row">
        <div class="col-md-6">
            <h3>Goal</h3>
            <div class="hline"></div>
            <p>The IQ-TREE software was created as the successor of <a href="http://www.cibiv.at/software/iqpnni/"><u>IQ</u>PNNI</a> and <a href="http://www.tree-puzzle.de"><u>TREE</u>-PUZZLE</a> (thus the name IQ-TREE). IQ-TREE was motivated by the rapid accumulation of phylogenomic data, leading to a need for efficient phylogenomic software that can handle a large amount of data and provide more complex models of sequence evolution.</p>
            <p>Therefore, we started to develop IQ-TREE in 2010 as open-source software under the <a href="http://www.gnu.org/licenses/licenses.en.html">GNU-GPL license</a>. Our aim is <em>(i)</em> to propose novel computational methods that perform better than existing approaches, <em>(ii)</em> to facilitate the inclusion of new maximum likelihood models that e.g. adequately capture more realistic aspects of sequence evolution, and <em>(iii)</em> to implement all commonly-used maximum likelihood analyses, allowing users to conduct many tasks using IQ-TREE.</p>
            <p>IQ-TREE is actively maintained by the core development team (see below) and a number of collabrators (see right).</p>

            <h3>Contact</h3>
            <div class="hline"></div>
            <p> If you have further questions, feedback, feature requests and bug reports, please sign up (if not done yet) and post a topic to <a href="https://groups.google.com/d/forum/iqtree">IQ-TREE Google group</a>.</p>
            <p><em>The average response time is one working day.</em></p>

            <h3>Credits and Acknowledgements</h3>
            <div class="hline"></div>
                <p>Some parts of the code were taken from the following packages/libraries: <a href="http://www.libpll.org">Phylogenetic likelihood library</a>, <a href="http://www.tree-puzzle.de">TREE-PUZZLE</a>, 
                <a href="http://dx.doi.org/10.1093/oxfordjournals.molbev.a025808">BIONJ</a>, <a href="http://dx.doi.org/10.1093/bioinformatics/btg319">Nexus Class Libary</a>, <a href="http://eigen.tuxfamily.org/">Eigen library</a>,
                <a href="http://www.sprng.org">SPRNG library</a>, <a href="http://www.zlib.net">Zlib library</a>, gzstream library, <a href="http://www.agner.org/optimize/">vectorclass library</a>, <a href="https://www.gnu.org/software/gsl/">GNU scientific library</a>.</p>
                <p>IQ-TREE was partially funded by the <a href="http://www.fwf.ac.at">Austrian Science Fund - FWF</a> (grants I 760-B17 from 2012-2015 and I 2508-B29 from 2016-2019) and the <a href="https://www.univie.ac.at">University of Vienna</a> (Initiativkolleg I059-N from 2011-2014).</p>
        </div><!--col-->
        <div class="col-md-6">
            <h3>Ongoing collaborations</h3>
            <div class="hline"></div>
            
            <h4>Phylogenetic likelihood library</h4>
            <p>Further development of the <a href="http://www.libpll.org">phylogenetic likelihood library</a> (<a href="http://dx.doi.org/10.1093/sysbio/syu084">Flouri et al., 2015</a>). In collaboration with <a href="http://sco.h-its.org/exelixis/people.html">Alexandros Stamatakis</a> and <a href="http://sco.h-its.org/exelixis/people.html">Tomas Flouri</a>.</p>

            <h4>Ultrafast parsimony bootstrap</h4>
            <p>An ultrafast approximation method for parsimony bootstrap. In collaboration with <a href="http://uet.vnu.edu.vn/~vinhls/">Le Sy Vinh</a> and <a href="http://www.cibiv.at/~diep_/">Diep Thi Hoang</a>.</p>
            
            <h4>Polymorphism Models (PoMo)</h4>
            <p>Efficient implementation of <a href="http://dx.doi.org/10.1093/sysbio/syv048">polymorphism aware model</a> into IQ-TREE. In collaboration with <a href="http://i122server.vu-wien.ac.at/pop/lab_members/Carolin_Kosiol.html">Carolin Kosiol</a> and <a href="http://i122server.vu-wien.ac.at/pop/lab_members/Dominik_Schrempf.html">Dominik Schrempf.</a></p>
            
            
            <h4>ModelFinder</h4>
            <p>A new model selection procedure based on IQ-TREE code. In collaboration with <a href="http://sydney.edu.au/science/biology/about_us/honorary_staff/jermiin_lars/">Lars Jermiin</a> and <a href="https://scholar.google.com/citations?user=zZT0fKUAAAAJ">Thomas Wong</a>.</p>

            <h4>Heterotachy models</h4>
            <p>Development of heterotachy models that allows for rate heterogeneity among lineages. In collaboration with <a href="http://www.utas.edu.au/maths-physics/people-old/profiles-mathematics/Barbara-Holland">Barbara Holland</a> and <a href="http://maths.adelaide.edu.au/people/stephen.crotty">Stephen Crotty</a>.</p>
            
            <h4>Non-reversible and Lie Markov models</h4>
            <p>In collaboration with <a href="http://www.utas.edu.au/profiles/staff/maths-physics/michael-woodhams">Michael Woodhams</a>, <a href="http://www.utas.edu.au/profiles/staff/maths-physics/jeremy-sumner">Jeremy Sumner</a>, and <a href="http://www.utas.edu.au/profiles/staff/maths-physics/Michael-Charleston">Michael Charleston</a>.</p>

            <h4>Approximate mixture models</h4>
            <p>In collaboration with <a href="http://rogerlab.biochemistryandmolecularbiology.dal.ca/Huaichun.php">Huaichun Wang</a>, <a href="http://rogerlab.biochemistryandmolecularbiology.dal.ca">Andrew Roger</a>, <a href="http://www.mathstat.dal.ca/~tsusko/">Edward Susko</a>.</p>
            
        </div>
    </div><! --/row -->
</div><! --/container -->