---
layout: post
title: "Dominik's paper accepted at <em>Journal of Theoretical Biology</em>"
author: Admin
categories: 
- news 
tags:
- publication
---

**D. Schrempf, B.Q. Minh, N. De Maio, A. von Haeseler, and C. Kosiol** (2016) Reversible polymorphism-aware phylogenetic models and their application to tree inference. *J. Theor. Biol.*, in press. 
<http://dx.doi.org/10.1016/j.jtbi.2016.07.042>

The associated IQ-TREE PoMo version and user guide are available at:

<http://www.iqtree.org/#variant>

**Highlights**

    - Species tree inference from genome-wide population data.
    - Takes incomplete lineage sorting into account.
    - Analytical solution of stationary distribution and formal proof of reversibility.
    - Reversibility ensures swiftness and stability.
    - Increase of sample size per species improves estimations without raising runtime.
    - Comparison to the Wright-Fisher diffusion. 

<!--more-->


>**Abstract:** We present a reversible Polymorphism-Aware Phylogenetic Model (revPoMo) for species tree estimation from genome-wide data. revPoMo enables the reconstruction of large scale species trees for many within-species samples. It expands the alphabet of DNA substitution models to include polymorphic states, thereby, naturally accounting for incomplete lineage sorting. We implemented revPoMo in the maximum likelihood software IQ-TREE. A simulation study and an application to great apes data show that the runtimes of our approach and standard substitution models are comparable but that revPoMo has much better accuracy in estimating trees, divergence times and mutation rates. The advantage of revPoMo is that an increase of sample size per species improves estimations but does not increase runtime. Therefore, revPoMo is a valuable tool with several applications, from speciation dating to species tree reconstruction.